common4j/src/main/com/microsoft/identity/common/java/net/UrlConnectionHttpClient.java

```diff
@@ -324,8 +319,15 @@ private HttpURLConnection setupConnection(HttpRequest request) throws IOExceptio
             urlConnection.setRequestProperty(entry.getKey(), entry.getValue());
         }
 
-        urlConnection.setRequestMethod(request.getRequestMethod());
+        if (urlConnection instanceof HttpsURLConnection) {
+            ((HttpsURLConnection) urlConnection).setSSLSocketFactory(
+                    SSLSocketFactoryWrapper.getSocketFactory(request.getSslContext())
```

Comment:
Can we let the SSL level be configured here?  We might want to enforce 1.3 at some point.

Reply:
@AdamBJohnsonx  do you mean enforce per request?

SSLSocketFactoryWrapper now supports TLS1.3, and it should be picked whenever the server supports it.

Reply:
My point is that we may want to be able to set a minimum.

Reply:
We could remove what we don't want to support from SSLSocketFactoryWrapper.SUPPORTED_SSL_PROTOCOLS.

That is... unless we specifically want to do this per-request.

(Of course, we'll have to kill off the non-https support as well).

Reply:
I'm thinking that this is likely a per-client decision.

Reply:
added a field in UrlConnectionHttpClientTest's builder.